src/components/Modal/Modal.tsx

```diff
@@ -75,7 +75,15 @@ export const Modal = ({
   };
 
   const handleEscKey = (e: KeyboardEvent<HTMLDivElement>) => {
```

Comment:
If you remove the `KeyboardEvent` from the import you could replace this with:

```suggestion
  const handleEscKey = (e: KeyboardEvent | React.KeyboardEvent<HTMLDivElement>) => {
```

This will make it easier to work with the event below.